packages/node-runtime-worker-thread/src/worker-runtime.spec.ts

```diff
@@ -40,23 +62,333 @@ describe('worker', () => {
+          '({ cls: class A{}, fn() {}, bsonType: NumberInt("1"), str: "123"})',
+          "{ cls: [class A], fn: [Function: fn], bsonType: Int32(1), str: '123' }"
+        ]
+      ];
+
+      primitiveValues.concat(everythingElse).forEach((testCase) => {
+        const [testName, evalValue, printable] = testCase;
+
+        it(testName, async() => {
+          const { init, evaluate } = caller;
+          await init('mongodb://nodb/', {}, { nodb: true });
+          const result = await evaluate(evalValue);
+          expect(result).to.have.property('printable');
+          if (printable instanceof RegExp) {
+            expect(result.printable).to.match(printable);
+          } else {
+            expect(result.printable).to.deep.equal(printable);
+          }
+        });
+      });
+    });
 
-      const result = await evaluate('1 + 1');
+    describe('shell-api results', () => {
+      const testServer = startTestServer('shared');
+      const db = `test-db-${Date.now().toString(16)}`;
+
+      type CommandTestRecord =
+        | [string | string[], string]
+        | [string | string[], string, any];
+
+      const showCommand: CommandTestRecord[] = [
+        [
+          'show dbs',
+          'ShowDatabasesResult',
+          ({ printable }: ShellResult) => {
+            expect(printable.find(({ name }: any) => name === 'admin')).to.not
+              .be.undefined;
+          }
+        ],
+        ['show collections', 'ShowCollectionsResult', []],
+        ['show profile', 'ShowProfileResult', { count: 0 }],
+        [
+          'show roles',
+          'ShowResult',
+          ({ printable }: ShellResult) => {
+            expect(printable.find(({ role }: any) => role === 'dbAdmin')).to.not
+              .be.undefined;
+          }
+        ]
+      ];
+
+      const useCommand: CommandTestRecord[] = [
+        [`use ${db}`, null, `switched to db ${db}`]
+      ];
+
+      const helpCommand: CommandTestRecord[] = [
+        [
+          'help',
+          'Help',
+          ({ printable }: ShellResult) => {
+            expect(printable).to.have.property('help', 'Shell Help');
+            expect(printable)
+              .to.have.property('docs')
+              .match(/https:\/\/docs.mongodb.com/);
+          }
+        ]
+      ];
+
+      const cursors: CommandTestRecord[] = [
+        [
+          [
+            `use ${db}`,
+            'db.coll.insertOne({ _id: ObjectId("000000000000000000000000"), foo: 321 });',
+            'db.coll.aggregate({ $match: { foo: 321 } })'
+          ],
+          'AggregationCursor',
+          ({ printable }: ShellResult) => {
+            expect(printable).to.have.property('cursorHasMore', false);
+            expect(printable)
+              .to.have.nested.property('[0]._id')
+              // TODO: chai assertion types
+              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+              // @ts-ignore
+              .bson(new ObjectId('000000000000000000000000'));
```

Comment:
I mean, if this is used only this time here i'd really prefer this assertion to be done without a plugin though, just doing the assertion here would be so much more clear.

Not to mention TS don't like that.

Reply:
Well, adding types for this is not an issue, if that's the only concern, I just legit forgot about this `todo` 😅 But you're right, it's used once, so I'll just refactor the test

Reply:
Done in 4bc1ae8